test/test-server/schema_registry_handlers.go

```diff
@@ -225,66 +225,64 @@ func handleSRById(t *testing.T) http.HandlerFunc {
 		idStr := vars["id"]
 		id64, err := strconv.ParseInt(idStr, 10, 32)
 		require.NoError(t, err)
-
-		schema := srsdk.Schema{Subject: "my-subject", Version: 1, Id: int32(id64)}
+		id32 := int32(id64)
+		schema := srsdk.Schema{Subject: srsdk.PtrString("my-subject"), Version: srsdk.PtrInt32(1), Id: srsdk.PtrInt32(id32)}
 		switch id64 {
 		case 1001:
-			schema.Schema = "schema0"
-			schema.References = []srsdk.SchemaReference{
+			schema.Schema = srsdk.PtrString("schema0")
+			schema.References = &[]srsdk.SchemaReference{
 				{
-					Name:    "ref_lvl1_1",
-					Subject: "lvl1-1",
-					Version: 1,
+					Name:    srsdk.PtrString("ref_lvl1_1"),
+					Subject: srsdk.PtrString("lvl1-1"),
+					Version: srsdk.PtrInt32(1),
 				},
 				{
-					Name:    "ref_lvl1_2",
-					Subject: "lvl1-2",
-					Version: 1,
+					Name:    srsdk.PtrString("ref_lvl1_2"),
+					Subject: srsdk.PtrString("lvl1-2"),
+					Version: srsdk.PtrInt32(1),
 				},
 			}
 		case 1002:
-			schema.Schema = "schema11"
-			schema.References = []srsdk.SchemaReference{
+			schema.Schema = srsdk.PtrString("schema11")
+			schema.References = &[]srsdk.SchemaReference{
 				{
-					Name:    "ref_lvl2",
-					Subject: "lvl2",
-					Version: 1,
+					Name:    srsdk.PtrString("ref_lvl2"),
+					Subject: srsdk.PtrString("lvl2"),
+					Version: srsdk.PtrInt32(1),
 				},
 			}
 		case 1003:
-			schema.Schema = "schema12"
-			schema.References = []srsdk.SchemaReference{
+			schema.Schema = srsdk.PtrString("schema12")
+			schema.References = &[]srsdk.SchemaReference{
 				{
-					Name:    "ref_lvl2",
-					Subject: "lvl2",
-					Version: 1,
+					Name:    srsdk.PtrString("ref_lvl2"),
+					Subject: srsdk.PtrString("lvl2"),
+					Version: srsdk.PtrInt32(1),
 				},
 			}
 		case 1004:
-			schema.Schema = "schema2"
-			schema.References = []srsdk.SchemaReference{}
+			schema.Schema = srsdk.PtrString("schema2")
+			schema.References = &[]srsdk.SchemaReference{}
 		case 1005:
-			schema.Schema = `{"schema":1}`
-			schema.References = []srsdk.SchemaReference{}
-			schema.Ruleset = &srsdk.RuleSet{
-				DomainRules: []srsdk.Rule{
-					{
-						Name: "checkSsnLen",
-						Kind: "CONDITION",
-						Mode: "WRITE",
-						Type: "CEL",
-						Expr: "size(message.ssn) == 9",
-					},
-				},
-			}
+			schema.Schema = srsdk.PtrString(`{"schema":1}`)
+			schema.References = &[]srsdk.SchemaReference{}
+			ruleSet := &srsdk.RuleSet{DomainRules: &[]srsdk.Rule{
+				{
+					Name: srsdk.PtrString("checkSsnLen"),
+					Kind: srsdk.PtrString("CONDITION"),
+					Mode: srsdk.PtrString("WRITE"),
+					Type: srsdk.PtrString("CEL"),
+					Expr: srsdk.PtrString("size(message.ssn) == 9"),
+				}}}
+			schema.Ruleset = *srsdk.NewNullableRuleSet(ruleSet)
```

Comment:
Any way we can combine these into one declaration?

Reply:
updated!